### PR TITLE
Task 3: Product service created with API Gateway and Lambdas

### PR DESCRIPTION
**Pull request by glhvta**
**Frontend application:** https://d3jv20c7ho3whs.cloudfront.net/

**Product service endpoints:**
  GET - https://bta8xv58vl.execute-api.eu-central-1.amazonaws.com/dev/products
  GET - https://bta8xv58vl.execute-api.eu-central-1.amazonaws.com/dev/products/{productId}

**Example**:
   - Product 200 response:   
   https://bta8xv58vl.execute-api.eu-central-1.amazonaws.com/dev/products/7567ec4b-b10c-45c5-9345-fc73c48a80a1
   - 404 Error:   
   https://bta8xv58vl.execute-api.eu-central-1.amazonaws.com/dev/products/random

**Swagger:**
  GET - https://dr4djol1jj.execute-api.eu-central-1.amazonaws.com/swagger
  GET - https://dr4djol1jj.execute-api.eu-central-1.amazonaws.com/swagger.json

**Task 3.1**

- [x] Create a lambda function called **getProductsList** under the same Serverless config file (i.e. serverless.yaml) of Product Service which will be triggered by the HTTP GET method.
- [x] The requested URL should be /products.
- [x] The response from the lambda should be a full array of products (mock data should be used - this mock data should be stored in Product Service).
- [x] This endpoint should be integrated with Frontend app for PLP (Product List Page) representation:  
 https://github.com/glhvta/shop-react-redux-cloudfront/pull/2

**Task 3.2**
- [x] Create a lambda function called **getProductsById** under the same Serverless config file (i.e. serverless.yaml) of Product Service which will be triggered by the HTTP GET method.
- [x] The requested URL should be /products/{productId} (what productId is in your application is up to you - productName, UUID, etc.).
- [x] The response from the lambda should be 1 searched product from an array of products (mock data should be used - this mock data should be stored in Product Service).
- [x] This endpoint is not needed to be integrated with Frontend right now.

**Task 3.3**
- [x] Commit all your work to separate branch (e.g. task-3 from the latest master) in your own repository.
- [x] Create a pull request to the master branch.

**Additional (optional) tasks**
- [x] **+1** - Async/await is used in lambda functions
- [x] **+1** - ES6 modules are used for Product Service implementation
- [x] **+1** - Custom Webpack/ESBuild/etc is manually configured for Product Service. Not applicable for preconfigured/built-in bundlers that come with templates, plugins, etc.
- [x] **+1** (All languages) - SWAGGER documentation is created for Product Service
- [x] **+1** (All languages) - Lambda handlers are covered by basic UNIT tests (NO infrastructure logic is needed to be covered)
- [x] **+1** (All languages) - Lambda handlers (getProductsList, getProductsById) code is written not in 1 single module (file) and separated in codebase.
- [x] **+1** All languages) - Main error scenarios are handled by API ("Product not found" error).

**Deadline:** 23.08.2022 / 23.08.2022

### CODE REVIEW

File: product-service/package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    "@typescript-eslint/eslint-plugin": "^5.34.0",
+    "@typescript-eslint/parser": "^5.34.0",
+    "esbuild": "^0.14.11",
+    "eslint": "^8.22.0",
+    "jest": "^28.1.3",
+    "json-schema-to-ts": "^1.5.0",
+    "serverless": "^3.0.0",
+    "serverless-auto-swagger": "^2.8.2",
+    "serverless-esbuild": "^1.23.3",
+    "serverless-webpack": "^5.8.0",
```
Comment (IgorByl) [inline product-service/package.json:30]: tried to use both esbuild and webpack =)

File: product-service/.gitignore [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# package directories
+node_modules
+jspm_packages
+
+# Serverless directories
+.serverless
+
+# esbuild directories
+.esbuild
```
Comment (IgorByl) [inline product-service/.gitignore]: .webpack ?

**Comment by IgorByl** [comment]
suggestion: `serverless-offline` package helps to speed up a local development

File: product-service/package.json (hunk 1 above)
Comment (glhvta) [reply product-service/package.json:30]: Yes, esbuild dependency is redundant here. Thanks!

File: product-service/.gitignore (hunk 2 above)
Comment (glhvta) [reply product-service/.gitignore]: updated

**Comment by glhvta** [comment]
> suggestion: `serverless-offline` package helps to speed up a local development

 Perfect! Will introduce in the scope of next PR.

 @IgorByl ,Thanks for the review!

**Pull request merged**